### PR TITLE
Feature/add githooks

### PR DESCRIPTION
**Pull request by annashtirberg**
this is for [Add git hooks for pre-commit flake8 #196](https://github.com/PythonFreeCourse/calendar/issues/196)
Used pre-commit for adding git hooks before doing commit
The hooks do flake8 checks like in the github actions
Also did yapf for making the code look good and fit flake8 style

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,7 +17,11 @@
 ## Creating development environment
 ### Prerequisites
 1. Windows or Linux based system - either [WSL on windows](https://docs.microsoft.com/en-us/windows/wsl/install-win10) or full blown linux.
-2. [Python](https://www.python.org/downloads/release/python-385/) 
+2. [Python](https://www.python.org/downloads/release/python-385/)
+3. Install pre-commit hooks:
+```shell
+pre-commit install
```
Comment (yammesicka) [inline README.md:23]: This comment doesn't work on my shell. Maybe it requires some another prerequisite?

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (annashtirberg) [reply README.md:27]: I changed it and I think it will work for you (and everyone else too) now :)

**Comment by annashtirberg** [comment]
The hooks need to be installed.
The way to do it is like this:
1. `pip install -r requirements.txt`  (install all requirements)
2. `pre-commit install` (install git hooks)

After this every commited file will be checked for style and also fixed if possible
If flake8 don't like your file the commit will fail and print flake8 errors to fix
If yapf changed your file the commit will fail and the file will need to "git add" again

Hope it will help everyone :)

File: app/routers/event.py [hunk 2]
```diff
@@ -29,34 +29,34 @@
...
 }
 
-
 router = APIRouter(
-    prefix="/event",
-    tags=["event"],
-    responses={404: {"description": "Not found"}},
+    prefix='/event',
+    tags=['event'],
+    responses={404: {
```
Comment (ivarshav) [inline app/routers/event.py:35]: I think you can keep it as one line

File: app/routers/event.py [hunk 3]
```diff
@@ -66,63 +66,71 @@ async def create_new_event(request: Request, session=Depends(get_db)):
...
-                                                      title, invited_emails)
+    uninvited_contacts = get_uninvited_regular_emails(session, owner_id, title,
+                                                      invited_emails)
 
     if is_zoom:
         raise_if_zoom_link_invalid(location)
 
-    event = create_event(session, title, start, end, owner_id, content,
-                         location, invited_emails, category_id=category_id)
+    event = create_event(session,
```
Comment (ivarshav) [inline app/routers/event.py:75]: please revert thie change

File: app/routers/event.py [hunk 4]
```diff
@@ -66,63 +66,71 @@ async def create_new_event(request: Request, session=Depends(get_db)):
...
-async def eventview(request: Request, event_id: int,
+@router.get('/{event_id}')
+async def eventview(request: Request,
+                    event_id: int,
                     db: Session = Depends(get_db)):
     event = by_id(db, event_id)
     start_format = '%A, %d/%m/%Y %H:%M'
-    end_format = ('%H:%M' if event.start.date() == event.end.date()
-                  else start_format)
+    end_format = ('%H:%M'
```
Comment (ivarshav) [inline app/routers/event.py:98]: I think the prev format was more readable

File: app/routers/event.py [hunk 5]
```diff
@@ -196,7 +201,11 @@ def update_event(event_id: int, event: Dict, db: Session
     return event_updated
 
 
-def create_event(db: Session, title: str, start, end, owner_id: int,
+def create_event(db: Session,
```
Comment (ivarshav) [inline app/routers/event.py:204]: Please revert this change

File: app/routers/event.py (hunk 2 above)
Comment (annashtirberg) [reply app/routers/event.py]: went back to original, sorry

File: app/routers/event.py (hunk 3 above)
Comment (annashtirberg) [reply app/routers/event.py]: went back to original, sorry

File: app/routers/event.py (hunk 4 above)
Comment (annashtirberg) [reply app/routers/event.py]: went back to original, sorry

File: app/routers/event.py (hunk 5 above)
Comment (annashtirberg) [reply app/routers/event.py]: went back to original, sorry

File: README.md (hunk 1 above)
Comment (yammesicka) [reply README.md:27]: image
^^"

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (annashtirberg) [reply README.md:27]: but I change it, ok now it is correct
you need to install the requirements before

Comment (yammesicka) [reply README.md:27]: Maybe you've changed the requirements.txt and forgot to push it?

Currently I don't have anything in the requirements.txt that installs the `pre-commit` command, unfortunately :(

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (annashtirberg) [reply README.md:27]: oh no, your right, my mistake
I fixed it now
